content/blog/probot/index.md

```diff
@@ -0,0 +1,83 @@
+  // Your code here
+  app.log('Yay, the app was loaded!')
+  const issue=context.payload.issue //getting issue details
+  const user=issue.login.user //Getting the username
+  app.on('issues.opened', async context =>({
+    return context.github.issues.createComment(context.issue({ body: `Thanks @${user} for opening this issue!:tada:
+    You are awesome..` })
+)
+  })
+}
+```
+The above code is for a bot which comments on new issues. It mentions the user too. The *"Robotic"* characteristics start from the 4th line starting with `app.on` function the first argument passed is the action which the bot listens to. These actions are **webhook** actions of GitHub and you can learn more about from[Webhook Documentation](https://developer.github.com/webhooks/).    
+Probot uses node's `async` API for returning the data. We used *`* to enclose the string/comment body because we won't need to use escape characters and we can also interpolate [Template Literals](https://flaviocopes.com/javascript-template-literals/).         
+
+Now we need to check if our app is working or not, we can run `npm run dev` and in the browser go to `localhost:8000`, you can find a webpage like this![probot-webpage](/wp.jpg)   
+ Click the **Register GitHub App** button and provide a name and install it in a repository.
+ > Tip: Create a new private repository because we'll be making a lot of noise for the testing   
+ 
+ Now, if you check the `.env`  file you'll see the variable values auto-filled with the necessary data. Glitch automatically deploys our app and our app will be listening to the webhooks to be received.
+ 
+ > Test the app by making a new issue in the test repo on GitHub.   
+ 
+ It works right?  
+Congratulations 🎉🎉 You just made yourself a GitHub App👌👌 
+##### Redelivering Webhook⤴️
+We can view all the webhooks of any event triggered by the Repository and we can redeliver them by going to **Settings->Developer Settings->GitHub Apps->"Your app name"->Advanced Tab** in your profile.
+ #### Commenting on PRs
+If you check the app settings in GitHub for your app, you can see that the app also works for Pull Requests. This is due to the edit on `app.yml` file before registering our app. 
+Now we need to edit the `index.js` to add this before the last '}'
+```js 
+//index.js
+app.on('pull_request.opened', async context =>{
+    const pr = context.payload.pull_request
+    const user = pr.user.login //Collecting Details of the person who created the PR
+     const msg=context.issue({body:`Hey @${user} :wave: Thanks for the PR !!! You are Awesome.`})
+     return context.github.issues.createComment(msg)
+  })
+```
+ Here we add a new _"robotic"_ characteristic by passing a webhook action called *pull_request.open*. Now our app listens for PR webhook and once received it emits the `context.issue({body})` to GitHub. You can see that we are still passing the body to the *createComment* function.This is because PRs are considered as another type of Issues in GitHub. 
```

Comment:
```suggestion
 Here we add a new _"robotic"_ characteristic by passing a webhook action called *pull_request.open*. Now our app listens for PR webhook and once received it emits the `context.issue({body})` to GitHub. You can see that we are still passing the body to the *createComment* function. This is because PRs are considered as another type of Issues in GitHub. 
```